### DISCUSSION TITLE
RPC Error[load_collection]

### DISCUSSION THREAD

**Discussion by VjDharnish** [milvus-io/milvus#26682] [Q&A and General discussion]
ISSUE
RPC error: [get_loading_progress], <MilvusException: (code=1, message=show collection failed: legacy error code:1, reason: load segment failed, OOM if load, maxSegmentSize = 555.8249225616455 MB, concurrency = 1, memUsage = 6687.296875 MB, predictMemUsage = 7243.1217975616455 MB, totalMem = 7851.5546875 MB thresholdFactor = 0.900000)>, <Time:{'RPC start': '2023-08-29 14:54:22.619762', 'RPC error': '2023-08-29 14:54:22.628156'}>

RPC error: [wait_for_loading_collection], <MilvusException: (code=1, message=show collection failed: legacy error code:1, reason: load segment failed, OOM if load, maxSegmentSize = 555.8249225616455 MB, concurrency = 1, memUsage = 6687.296875 MB, predictMemUsage = 7243.1217975616455 MB, totalMem = 7851.5546875 MB thresholdFactor = 0.900000)>, <Time:{'RPC start': '2023-08-29 14:47:18.840019', 'RPC error': '2023-08-29 14:54:22.634163'}>

RPC error: [load_collection], <MilvusException: (code=1, message=show collection failed: legacy error code:1, reason: load segment failed, OOM if load, maxSegmentSize = 555.8249225616455 MB, concurrency = 1, memUsage = 6687.296875 MB, predictMemUsage = 7243.1217975616455 MB, totalMem = 7851.5546875 MB thresholdFactor = 0.900000)>, <Time:{'RPC start': '2023-08-29 14:47:18.836502', 'RPC error': '2023-08-29 14:54:22.634920'}>

 I am Currently Facung this. issue while loading a collection with entries of 1000000  in MIlvus Standalone. 
in This , I Created 10 collections each having same datasets.  upto. 8 collections run smoothly gives the expected result
But for 9th and 10th collection , It shows the above Error

**Comment by yhmo** [comment]
"OOM if load ..... memUsage = 6687.296875 MB ....... totalMem = 7851.5546875 MB thresholdFactor = 0.900000"
The total RAM is 8GB, currently memory usage is 6.6GB, each segment size is 500MB+. If this segment is loaded, the total usage might exceed the threashold 0.9*8GB=7.2GB, to avoid potential OOM risk, Milvus deny to load the segment. This behavior is as-designed.

You need to enlarge the RAM size for this container.

**Reply by VjDharnish** [reply]
Hi,thank you for the response , I understood this But as I mentioned , Each collection has the same dataset upon this, out of 10 , 8 collections are loaded without any flaws,  But when come to the last 9 and 10th collection it shows above error , Why if all have same data inside it then why these 2 only take huge amount of RAM 

I also did Droping all Collections and indexes and recreated it , this time my 5th and 6th collection show that error after  runnning above 10 minutes? Help me on this

**Reply by yhmo** [reply]
When it does loading, for each segment, it estimates memory usage by the current segment size. Typically, if `current_segment_size*3 + current_memory_usage > total_ram*threashold`, it will return this error. So, the current segment size is the key point.

Although the same dataset is in each collection, the number of segments might be different.
For example:
collection_1 has 3 segments: 512MB,  256MB, 256MB, total size is 1GB
collection_2 has 2 segments: 512MB,  512MB, total size is 1GB

Assume the total RAM is 8GB, and current memory usage is 6GB. threshold is 0.9
When you call collection_1.load(), the last segment is 256MB, the estimated memory usage is 256MB*3. So,  256MB*3+6GB < 8GB*0.9
When you call collection_1.load(), the last segment is 256MB, the estimated memory usage is 512MB*3. So,  512MB*3+6GB > 8GB*0.9

This is why you get this error.
